mlir/lib/Quantum/Transforms/ScatterPatterns.cpp

```diff
@@ -217,28 +304,34 @@ struct ScatterOpRewritePattern : public mlir::OpRewritePattern<mhlo::ScatterOp>
 
     void generateIndicesRecursive(const std::vector<int64_t> &shape,
                                   std::vector<int64_t> &currentIndex, int64_t dimension,
-                                  std::vector<std::vector<int64_t>> &configurations) const
+                                  SmallVector<Value> &configurations,
+                                  mlir::PatternRewriter &rewriter, Location loc) const
     {
         if (dimension == shape.size()) {
-            configurations.push_back(currentIndex);
+            for (auto elem : currentIndex) {
+                // integer to Value
+                auto valueCurrentIndex = rewriter.create<index::ConstantOp>(loc, elem);
+                // Add to configuration
+                configurations.push_back(valueCurrentIndex);
+            }
         }
         else {
             for (int i = 0; i < shape[dimension]; i++) {
                 currentIndex.push_back(i);
-                generateIndicesRecursive(shape, currentIndex, dimension + 1, configurations);
+                generateIndicesRecursive(shape, currentIndex, dimension + 1, configurations,
+                                         rewriter, loc);
                 currentIndex.pop_back();
             }
         }
     }
 
-    SmallVector<Value> getResultsIndices(std::vector<int64_t> updatesIndices,
-                                         std::vector<int64_t> updateScatterIndices,
-                                         std::vector<int64_t> updateWindowsIndices,
+    SmallVector<Value> getResultsIndices(SmallVector<Value> updateScatterIndices,
```

Comment:
(Especially) since this code is quite complex, it would be good to have some docstrings that explain what these three helper functions are doing conceptually.

Reply:
Added more functions and added some docs :+1: